internal/pkg/instrumentation/bpf/net/http/client/probe.go

```diff
@@ -79,7 +79,7 @@ func New(logger logr.Logger) probe.Probe {
 			},
 		},
 		Uprobes: map[string]probe.UprobeFunc[bpfObjects]{
-			"net/http.(*Client).do": uprobeDo,
+			"net/http.(*Transport).roundTrip": uprobeDo,
```

Comment:
Probably better to rename `uprobeDo` to something like `uprobeRoundTrip`.
In addition, I think it is worth renaming the probe name in the `bpf.probe.c` file